internal/cli/run/run.go

```diff
@@ -13,6 +16,20 @@ import (
 	"github.com/ooni/probe-cli/nettests"
 )
 
+// listenForSignals will listen for SIGINT and SIGTERM. When it receives those
+// signals it will set isTerminated to true, which will cleanly shutdown the
+// test logic
+func listenForSignals(ctx *ooni.Context) {
+	s := make(chan os.Signal, 1)
+	signal.Notify(s, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-s
+		log.Debugf("caught a signal, shutting down cleanly")
+		ctx.IsTerminated = true
```

Comment:
Please, use `atomic` to make sure we're accessing the variable with atomic semantics. I am thinking at something like `atomic.AddInt64(&ctx.IsTerminated)` to stop and `atomic.LoadInt64(&ctx.IsTerminated)` to fetch it.

Unfortunately there is no atomic boolean. If you wish to retain a boolan type, otherwise use a `sync.Mutex` but I have a mild preference for atomic semantics because it's more robust than explicitly using mutexes.